src/clip.sol

```diff
@@ -281,12 +284,14 @@ contract Clipper {
         }
 
         // Calculate remaining lot after operation
+        // TODO: could we end up with 1 wei rounding error?
         sale.lot = sale.lot - slice;  // safe because: slice <= lot
 
         // Send collateral to who
         vat.flux(ilk, address(this), who, slice);
 
         // Do external call (if defined)
+        // TODO: do we want to circuit break flash loans?
```

Comment:
I feel that we shouldn't do it, but not really opposed to have it if the rest prefers so.

Reply:
I'm working on it already: https://github.com/makerdao/dss/pull/182

Why not? I think it is by far the riskiest part of the whole code. Also it has zero unit tests rn.

I would even go as far as to suggest to launch liquidations 2.0 with external calls disabled, and only enable them when we make sure everything else works correctly.

Reply:
It's one of the most interesting functionalities of auctions 2.0, so I wouldn't leave it out for the launch at all.
Technically any part of the code might have a bug and we are not adding functionality to circuit break them.
I agree that external calls are riskier but I'd say we better put the effort to make sure it is working fine and malicious calls attempts can't do anything to it.

Reply:
@gbalabasquer OK. Instead I added tests for external calls: https://github.com/makerdao/dss/pull/185